### PR TITLE
on startup creation of maps

### PR DESCRIPTION
**Pull request by greed2411**
for pulling recent `Config`-like structures for all the available projects (`project_name`s). on startup, builder-backend is pinged to collect all such configs and stored globally.

### CODE REVIEW

**Comment by greed2411** [comment]
as per contracts discussed for creation, has been followed. please review and merge it.
for testing it locally, try running it with `BUILDER_BACKEND_URL='http://builder.vernacular.ai' py.test -v` within the `slu` directory.

File: slu/slu/utils/config.py [hunk 1]
```diff
@@ -393,3 +375,49 @@ def remove_checkpoints(self, task) -> None:
...
+
+    def _get_configs_from_api(self):
+
+        BUILDER_BACKEND_URL = os.getenv("BUILDER_BACKEND_URL")
+        if BUILDER_BACKEND_URL is None:
+            raise ValueError(
+                f"missing BUILDER_BACKEND_URL env variable, please set it appropriately."
+            )
+        url = BUILDER_BACKEND_URL + const.CLIENTS_CONFIGS_ROUTE
+        for _ in range(5):
```
Comment (ltbringer) [inline slu/slu/utils/config.py:408]: Requests have [code for retries](https://stackoverflow.com/questions/15431044/can-i-set-max-retries-for-requests-request).

File: slu/tests/test_config/test_config_creation.py [hunk 2]
```diff
@@ -52,20 +51,33 @@ def test_api_config_json():
...
-        os.getenv("BUILDER_BACKEND_URL"),
-        body=json.dumps(actual_config)
+        url,
+        body=json.dumps(body_resp)
     )
 
-    json_config_data_provider = JSONAPIConfigDataProvider()
-    config = Config(config_data_provider=json_config_data_provider)
-    assert actual_config == config._config
+    client_configs = {
```
Comment (ltbringer) [inline slu/tests/test_config/test_config_creation.py:74]: Should this be here?

File: slu/tests/test_config/test_config_creation.py [hunk 3]
```diff
@@ -52,20 +51,33 @@ def test_api_config_json():
     import os
     import json
 
-    from slu.utils.config import Config, JSONAPIConfigDataProvider
+    from slu.utils.config import Config, OnStartupClientConfigDataProvider
+    from slu import constants as const
     
     with open("config/config.json") as handler:
-        actual_config = json.load(handler)
+        axis_config = json.load(handler)
```
Comment (ltbringer) [inline slu/tests/test_config/test_config_creation.py:58]: I am scared of 57-63.

File: slu/tests/test_config/test_config_creation.py [hunk 4]
```diff
@@ -52,20 +51,33 @@ def test_api_config_json():
...
+
+    with open("config/oyo.json") as handler:
+        oyo_config = json.load(handler)
+
+    body_resp = [axis_config, oyo_config]
+
+    BUILDER_BACKEND_URL = "http://builder.vernacular.ai"
```
Comment (ltbringer) [inline slu/tests/test_config/test_config_creation.py:65]: env vars are nice.

**Review by ltbringer — CHANGES_REQUESTED** [review]

File: slu/tests/test_config/test_config_creation.py (hunk 3 above)
Comment (greed2411) [reply slu/tests/test_config/test_config_creation.py:58]: why tho? assumption of files being present/missing?

File: slu/tests/test_config/test_config_creation.py (hunk 2 above)
Comment (greed2411) [reply slu/tests/test_config/test_config_creation.py:74]: not following. i think it works as a test, idk if you are looking at something else.

File: slu/tests/test_config/test_config_creation.py (hunk 4 above)
Comment (ltbringer) [reply slu/tests/test_config/test_config_creation.py:65]: Missed that these are tests, resolving.

File: slu/slu/utils/config.py (hunk 1 above)
Comment (greed2411) [reply slu/slu/utils/config.py]: fixed it with `urllib3` based `Retry` along with `requests` based `HTTPAdapter`

**Review by ltbringer — APPROVED** [review]

**Pull request merged**